### PR TITLE
Enhanced messaging and error notification for update_migration_status method.

### PR DESCRIPTION
**Pull request by billfitzgerald0120**
Add log message for errors.
Refactor to new-style method.
Create new-style spec test.

https://www.pivotaltracker.com/story/show/146653755

@miq-bot add_label refactoring, test, fine/yes

### CODE REVIEW

**Comment by billfitzgerald0120** [comment]
@tinaafitz 
Please Review

File: content/automate/ManageIQ/Infrastructure/VM/Migrate/StateMachines/VMMigrate.class/__methods__/update_migration_status.rb [hunk 1]
```diff
@@ -1,18 +1,66 @@
...
+          module StateMachines
+            class UpdateMigrationStatus
+              def initialize(handle = $evm)
+                @handle = handle
+              end
 
-miq_task = $evm.root['vm_migrate_task']
-status   = $evm.inputs['status']
+              def main
+                @handle.log(:info, "Starting update_migration_status")
```
Comment (tinaafitz) [inline content/automate/ManageIQ/Infrastructure/VM/Migrate/StateMachines/VMMigrate.class/__methods__/update_migration_status.rb]: @billfitzgerald0120 Can we remove the starting log message? The Automate engine creates a method starting log message.

File: content/automate/ManageIQ/Infrastructure/VM/Migrate/StateMachines/VMMigrate.class/__methods__/update_migration_status.rb [hunk 2]
```diff
@@ -1,18 +1,66 @@
...
+                update_migration_status(task)
+                @handle.log(:info, "Ending update_migration_status")
```
Comment (tinaafitz) [inline content/automate/ManageIQ/Infrastructure/VM/Migrate/StateMachines/VMMigrate.class/__methods__/update_migration_status.rb]: @billfitzgerald0120 Can we remove the ending log message?

File: content/automate/ManageIQ/Infrastructure/VM/Migrate/StateMachines/VMMigrate.class/__methods__/update_migration_status.rb [hunk 3]
```diff
@@ -1,18 +1,66 @@
...
+                raise "ERROR - task object not passed in" unless @handle.root['vm_migrate_task']
+                @handle.root['vm_migrate_task']
+              end
+
+              def update_migration_status(task)
+                status = @handle.inputs['status']
+
+                # Update Status Message
+                updated_message  = "Server [#{@handle.root['miq_server'].name}] "
+                # updated_message += "VM [#{task.get_option(:vm_target_name)}] "
```
Comment (tinaafitz) [inline content/automate/ManageIQ/Infrastructure/VM/Migrate/StateMachines/VMMigrate.class/__methods__/update_migration_status.rb]: @billfitzgerald0120 Do we have the vm_target_name?

File: spec/content/automate/ManageIQ/Infrastructure/VM/Migrate/StateMachines/VMMigrate.class/__methods__/update_migration_status_spec.rb [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  let(:miq_server) { EvmSpecHelper.local_miq_server }
+  let(:user) { FactoryGirl.create(:user_with_email_and_group) }
+  let(:miq_request_task) do
+    FactoryGirl.create(:miq_request_task, :miq_request => request,
+                       :source => vm, :state => 'fred')
+  end
+  let(:request) { FactoryGirl.create(:vm_migrate_request, :requester => user) }
+  let(:ems) { FactoryGirl.create(:ems_vmware, :zone => FactoryGirl.create(:zone)) }
+  let(:vm) { FactoryGirl.create(:vm_vmware, :ems_id => ems.id, :evm_owner => user) }
+  let(:svc_model_vm) do
```
Comment (tinaafitz) [inline spec/content/automate/ManageIQ/Infrastructure/VM/Migrate/StateMachines/VMMigrate.class/__methods__/update_migration_status_spec.rb]: @billfitzgerald0120 I don't think svc_model_vm is used.

File: spec/content/automate/ManageIQ/Infrastructure/VM/Migrate/StateMachines/VMMigrate.class/__methods__/update_migration_status_spec.rb [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    MiqAeMethodService::MiqAeServiceVm.find(vm.id)
+  end
+  let(:svc_model_miq_server)       { MiqAeMethodService::MiqAeServiceMiqServer.find(miq_server.id) }
+  let(:svc_model_miq_request_task) { MiqAeMethodService::MiqAeServiceMiqRequestTask.find(miq_request_task.id) }
```
Comment (tinaafitz) [inline spec/content/automate/ManageIQ/Infrastructure/VM/Migrate/StateMachines/VMMigrate.class/__methods__/update_migration_status_spec.rb]: @billfitzgerald0120 I don't think svc_model_miq_request_task is used.

File: spec/content/automate/ManageIQ/Infrastructure/VM/Migrate/StateMachines/VMMigrate.class/__methods__/update_migration_status_spec.rb [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+      current_object = Spec::Support::MiqAeMockObject.new
+      current_object.parent = root_object
+      service.object = current_object
+    end
+  end
+
+  shared_examples_for "#Update Migration Status" do
+    it "check" do
+      ae_service.root['ae_result'] = ae_result
+      allow(ae_service).to receive(:inputs) { {'status' => "fred"} }
```
Comment (tinaafitz) [inline spec/content/automate/ManageIQ/Infrastructure/VM/Migrate/StateMachines/VMMigrate.class/__methods__/update_migration_status_spec.rb]: @billfitzgerald0120 I see that you're passing status into the shared example. Should status here be the passed in status?

**Comment by billfitzgerald0120** [comment]
@tinaafitz Made changes as requested, please review

File: content/automate/ManageIQ/Infrastructure/VM/Migrate/StateMachines/VMMigrate.class/__methods__/update_migration_status.rb [hunk 7]
```diff
@@ -1,18 +1,64 @@
...
+
+                # Update Status Message
+                updated_message  = "Server [#{@handle.root['miq_server'].name}] "
+                updated_message += "VM [#{task.source.name}] "
+                updated_message += "Step [#{@handle.root['ae_state']}] "
+                updated_message += "Status [#{status}] "
+                updated_message += "Message [#{task.message}] "
+                updated_message += "Current Retry Number [#{@handle.root['ae_state_retries']}]" if @handle.root['ae_result'] == 'retry'
+                task.miq_request.user_message = updated_message
+                task.message = status
```
Comment (mkanoor) [inline content/automate/ManageIQ/Infrastructure/VM/Migrate/StateMachines/VMMigrate.class/__methods__/update_migration_status.rb]: @billfitzgerald0120 
Is the same as line 51

File: content/automate/ManageIQ/Infrastructure/VM/Migrate/StateMachines/VMMigrate.class/__methods__/update_migration_status.rb [hunk 8]
```diff
@@ -1,18 +1,64 @@
...
+                  @handle.create_notification(:level   => "error", \
+                                              :message => "VM Migration Error: #{updated_message}")
+                  @handle.log(:error, "VM Migration Error: #{updated_message}")
+                end
+
+                # Update Status for on_entry,on_exit
+                if @handle.root['ae_result'] == 'ok'
+                  if status == 'migration_complete'
+                    message = 'VM Migrated Successfully'
+                    task.finished(message)
```
Comment (mkanoor) [inline content/automate/ManageIQ/Infrastructure/VM/Migrate/StateMachines/VMMigrate.class/__methods__/update_migration_status.rb]: @billfitzgerald0120 we dont need the message variable we can directly call
``` ruby
task.finished('VM Migrated Successfully')
```

Comment (billfitzgerald0120) [inline content/automate/ManageIQ/Infrastructure/VM/Migrate/StateMachines/VMMigrate.class/__methods__/update_migration_status.rb]: Made changes as requested

**Comment by simaishi** [comment]
@billfitzgerald0120 Is there a BZ for this?  Can you please create if it doesn't exist?

**Comment by billfitzgerald0120** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=1478462

**Comment by billfitzgerald0120** [comment]
@simaishi I created a BZ for this https://bugzilla.redhat.com/show_bug.cgi?id=1478462

**Comment by simaishi** [comment]
Fine backport details:

```
$ git log -1
commit b8bfb5d7602baba2ae2aa11fba5a71ef180b29e6
Author: Madhu Kanoor <mkanoor@redhat.com>
Date:   Thu Jun 29 16:50:13 2017 -0400

    Merge pull request #125 from billfitzgerald0120/vm_migrate
    
    Enhanced messaging and error notification for update_migration_status method.
    (cherry picked from commit 7a789402519c390e8b21c64f0fe5566f4987ff54)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1478520
```

**Pull request merged**